### PR TITLE
feat(dbss): add new resource to manage RDS database bind DBSS instance

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add new resource manages add RDS database to DBSS instance.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support new resource manages add RDS database to DBSS instance.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/dbss" TESTARGS="-run TestAccAddRdsDatabase_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss -v -run TestAccAddRdsDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccAddRdsDatabase_basic
=== PAUSE TestAccAddRdsDatabase_basic
=== CONT  TestAccAddRdsDatabase_basic
--- PASS: TestAccAddRdsDatabase_basic (906.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      906.756s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image
    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

### CODE REVIEW

File: docs/resources/dbss_rds_database.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+subcategory: "Database Security Service (DBSS)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_dbss_rds_database"
+description: |-
+  Manages add an RDS database to DBSS instance within HuaweiCloud.
```
Comment (jinyangyang222) [inline docs/resources/dbss_rds_database.md:6]: Manage the resource of adding RDS database to DBS instance within HuaweiCloud.

File: docs/resources/dbss_rds_database.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the DBSS instance ID.
+  Changing this parameter will create a new resource.
+
+* `rds_id` - (Required, String, ForceNew) Specifies the RDS instance ID.
```
Comment (jinyangyang222) [inline docs/resources/dbss_rds_database.md:39]: RDS instance ID or RDS databases ID?

File: docs/resources/dbss_rds_database.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  + **OFF**
+
+  After an RDS database is associated with the DBSS instance, the audit status is **OFF** by default.
+
+* `lts_audit_switch` - (Optional, Int) Specifies whether to disable LTS audit.
+  The valid values are as follows:
+  + **1**: Indicates disable.
+  + **0**: Remain unchanged. (In this case, the value can also be an integer other than **1**).
+
+  -> This parameter is used in the DWS database scenario. If you do not disable LTS audit, no operation is required.
```
Comment (jinyangyang222) [inline docs/resources/dbss_rds_database.md:69]:
```suggestion
  -> This parameter is used in the DWS database scenario. If you do not need to close it, there is no need to set this field.
```

File: docs/resources/dbss_rds_database.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  The valid values are as follows:
+  + **ON**
+  + **OFF**
+
+  After an RDS database is associated with the DBSS instance, the audit status is **OFF** by default.
+
+* `lts_audit_switch` - (Optional, Int) Specifies whether to disable LTS audit.
+  The valid values are as follows:
+  + **1**: Indicates disable.
+  + **0**: Remain unchanged. (In this case, the value can also be an integer other than **1**).
```
Comment (deer-hang) [inline docs/resources/dbss_rds_database.md]:
```suggestion
  + `1`: Indicates disable.
  + `0`: Remain unchanged. (In this case, the value can also be an integer other than `1`).
```

File: huaweicloud/services/acceptance/dbss/resource_huaweicloud_dbss_rds_database_test.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+}
+`, testAccAddRdsDatabase_base(name))
+}
+
+func testAccAddRdsDatabaseImportState(rName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		var instanceId, rdsId string
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", rName, rs)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dbss/resource_huaweicloud_dbss_rds_database_test.go]: Variable `rs` is always nil in this line.
```suggestion
			return "", fmt.Errorf("resource (%s) not found", rName)
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+		return diag.Errorf("unable to find the RDS instance ID from the API response")
+	}
+
+	d.SetId(resourceId)
+
+	status := d.Get("status").(string)
+	if status == "ON" {
+		databaseInfo, err := GetDatabaseList(client, instanceId, d.Id())
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving databases")
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]: I think the `checkDeleted` function should not appear here.

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+		return diag.Errorf("error creating DBSS client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	databaseId := d.Get("db_id").(string)
+	if databaseId == "" {
+		diag.Errorf("failed to updating the audit status: 'db_id' is not found in API response")
+	}
+
+	if d.HasChange("status") {
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]: The field `lts_audit_switch` was forgotten.

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("dbss", region)
+	if err != nil {
+		return diag.Errorf("error creating DBSS client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	databaseId := d.Get("db_id").(string)
+	if databaseId == "" {
+		diag.Errorf("failed to updating the audit status: 'db_id' is not found in API response")
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
		diag.Errorf("edit audit status is not currently supported: 'db_id' is not found in API response")
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resourceId := utils.PathSearch("ret_list[0].id", respBody, "").(string)
+	if resourceId == "" {
+		return diag.Errorf("unable to find the RDS instance ID from the API response")
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
		return diag.Errorf("error adding RDS database to the DBSS instance: ID is not found in API response")
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go (hunk 6 above)
Comment (deer-hang) [reply huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
			return diag.Errorf("error retrieving DBSS audit databases in creation")
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+
+	status := d.Get("status").(string)
+	if status == "ON" {
+		databaseInfo, err := GetDatabaseList(client, instanceId, d.Id())
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving databases")
+		}
+		databaseId := utils.PathSearch("database.id", databaseInfo, "").(string)
+		if databaseId == "" {
+			diag.Errorf("failed to enable the audit status: 'db_id' is not found in API response")
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
			return diag.Errorf("failed to enable the audit status: 'db_id' is not found in API response")
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("dbss", region)
+	if err != nil {
+		return diag.Errorf("error creating DBSS client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	databaseInfo, err := GetDatabaseList(client, instanceId, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving databases")
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error retrieving DBSS audit databases")
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+		d.Set("ip", utils.PathSearch("database.ip", databaseInfo, nil)),
+		d.Set("port", utils.PathSearch("database.port", databaseInfo, nil)),
+		d.Set("os", utils.PathSearch("database.os", databaseInfo, nil)),
+		d.Set("audit_status", utils.PathSearch("database.audit_status", databaseInfo, nil)),
+		d.Set("agent_url", utils.PathSearch("database.agent_url", databaseInfo, nil)),
+		d.Set("db_classification", utils.PathSearch("database.db_classification", databaseInfo, nil)),
+		d.Set("rds_audit_switch_mismatch", utils.PathSearch("database.rds_audit_switch_mismatch", databaseInfo, nil)),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go:214]: return diag.FromErr(mErr.ErrorOrNil())

Comment (deer-hang) [inline huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]:
```suggestion
	return diag.FromErr(mErr.ErrorOrNil())
```

File: huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go (hunk 8 above)
Comment (deer-hang) [reply huaweicloud/services/dbss/resource_huaweicloud_dbss_rds_database.go]: Remove this line to line 301.

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dbss" TESTARGS="-run TestAccAddRdsDatabase_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss -v -run TestAccAddRdsDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccAddRdsDatabase_basic
=== PAUSE TestAccAddRdsDatabase_basic
=== CONT  TestAccAddRdsDatabase_basic
--- PASS: TestAccAddRdsDatabase_basic (1103.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      1103.972s
```

**Pull request merged**